### PR TITLE
Database utility mixins

### PR DESCRIPTION
**Pull request by revmischa**
Move over QueryFilter, SoftDeletable, ExtID, Upsertable from sls-flask.
Augment BaseQuery.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -47,6 +47,8 @@ black = "*"
 pytest-mock = "*"
 pytest-mypy = "*"
 moto = "*"
+ipython = "*"
+prompt-toolkit = "==2.0.10"
```
Comment (adamszeptycki) [inline Pipfile]: why would that be another dependency?

File: jetkit/db/extid.py [hunk 2]
```diff
@@ -16,17 +18,33 @@ class ExtID(Generic[T]):
...
+        index=True,
     )
 
     @classmethod
-    def get_by_extid(cls: T, uuid: str) -> Optional[T]:
-        return cls.query.filter_by(extid=uuid).one_or_none()
+    def get_by_extid(cls: Type[db.Model], uuid: str) -> Optional[T]:
+        try:
+            return cls.query.filter_by(extid=uuid).one_or_none()
+        except DataError:  # This means that extid is not a valid uuid
```
Comment (adamszeptycki) [inline jetkit/db/extid.py]: heh? not valid you mean resource with that type is not in database.

File: jetkit/db/query_filter.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        This is not done via a query so it must be a code-level check.
+        """
+        raise NotImplementedError()
+
+    @property
+    def entity(self):
+        return self._primary_entity.type
+
+
+class FilteredQuery(QueryFilter):
```
Comment (adamszeptycki) [inline jetkit/db/query_filter.py]: is that for SoftDeleted and similar stuff?

File: Pipfile (hunk 1 above)
Comment (revmischa) [reply Pipfile]: https://github.com/ipython/ipython/issues/11975

File: jetkit/db/extid.py (hunk 2 above)
Comment (revmischa) [reply jetkit/db/extid.py]: `ERROR:  invalid input syntax for type uuid: "foo"`

File: jetkit/db/query_filter.py (hunk 3 above)
Comment (revmischa) [reply jetkit/db/query_filter.py]: Yes, the idea is to let us stack those kinds of query classes without the mixins overwriting each other

**Pull request merged**